query-engine/core/src/interactive_transactions/actors.rs

```diff
@@ -361,13 +328,19 @@ pub fn spawn_itx_actor(
 */
 pub fn spawn_client_list_clear_actor(
     clients: Arc<RwLock<HashMap<TxId, ITXClient>>>,
+    closed_txs: Arc<RwLock<lru::LruCache<TxId, ()>>>,
     mut rx: Receiver<TxId>,
 ) -> JoinHandle<()> {
     tokio::task::spawn(async move {
         loop {
             if let Some(id) = rx.recv().await {
                 trace!("removing {} from client list", id);
-                clients.write().await.remove(&id);
+
+                let mut guard = clients.write().await;
```

Comment:
Another minor nit. This is a guard and the actual Hashmap. So a better name is `client_map` or `map`.

Reply:
Renamed it to `clients_guard`, since we're dropping the guard, not the map, right?